felix/nftables/ipsets.go

```diff
@@ -383,8 +383,15 @@ func (s *IPSets) tryResync() error {
 	setsChan := make(chan setData)
 	defer close(setsChan)
 
-	// Start a goroutine to list the elements of each set.
+	// Start a goroutine to list the elements of each set. Limit to 100 concurrent set reads to
+	// avoid spawning too many goroutines if there are a large number of sets.
+	routineLimit := make(chan struct{}, 100)
```

Comment:
I might have used `runtime.GOMAXPROCS()` to pick a limit that scales with CPU.

Reply:
Still digging into this one - for some reason setting this to GOMAXPROCS caused a lot of tests to fail.